### DISCUSSION TITLE
Resize like yabai

### DISCUSSION THREAD

**Discussion by HCHogan** [nikitabobko/AeroSpace#1007] [feature-ideas]
The current resize behavior is a little weird to me, can we add a new resize feature that moves the separate line between windows instead of just fixed on increase/decrease the current window size?
for example like this in yabai:
```bash
lctrl + alt - h : yabai -m window --resize left:-50:0; \
                  yabai -m window --resize right:-50:0
lctrl + alt - j : yabai -m window --resize bottom:0:50; \
                  yabai -m window --resize top:0:50
lctrl + alt - k : yabai -m window --resize top:0:-50; \
                  yabai -m window --resize bottom:0:-50
lctrl + alt - l : yabai -m window --resize right:50:0; \
                  yabai -m window --resize left:50:0

```

**Comment by marshes** [comment]
could work like this:

resize [-h|--help] [--window-id <window-id>] (smart|smart-opposite|width|height|left|right|top|bottom) [+|-]<number>

if specifying a side, change the side of the active window by <number>

this would allow for keeping the same functionality, but also adding yabai functionality as well. I glanced at the code to see if I could figure it out but it was beyond me.

Edit, looks like 2 files can be modified to add additional resize cases: ResizeCmdArgs.swift to add left | right | top | bottom to the parser and then define cases in ResizeCommand.swift. I have a local debug build working, just need to do some comparison to how yabai handles resizing to make the cases match.

**Comment by marshes** [comment]
Cool, I leveraged a bit of help from Claude and have a local build tested and it's working well, so much so that I might use the debug build as standard on my machine, as it seems to be working well. My changes can be seen in my fork.

Relevant toml config I'm currently testing which shifts the window/node by 100 for j/l/i/k keys, and expands or shrinks the size for m/n keys. Note, if the window shares a side with the edge of your monitor, shifting a window effectively allows you to expand or shrink the window similar to resize, however I find "shifting" to be more intuitive as the direction the window changes is consistent with the button press, regardless of which side of the monitor the window is located. 

```
ctrl-shift-j = ['resize left +100', 'resize right -100']
ctrl-shift-l = ['resize left -100', 'resize right +100']
ctrl-shift-i = ['resize top +100', 'resize bottom -100']
ctrl-shift-k = ['resize top -100', 'resize bottom +100']
ctrl-shift-m = 'resize smart +100'
ctrl-shift-n = 'resize smart -100'
ctrl-shift-b = 'balance-sizes'
```

I'm not too comfortable with Swift, so I don't fully understand some of the changes made to the ResizeCommand file.

Will look into doing a pull request this weekend -though also not too familiar with how to do that either. No better way to learn I suppose...

Hope this helps someone!

**Comment by marshes** [comment]
My fork seems to be working well. I haven't updated the test cases for the new single sided shifts, but local testing seems to work properly. @HCHogan feel free to build my fork and give it a spin. Can always pull it into mainline or a separate branch, but would request some guidance!
